pallets/creditcoin/src/migrations/mod.rs

```diff
@@ -9,55 +17,30 @@ mod v5;
 mod v6;
 mod v7;
 
-#[cfg(feature = "try-runtime")]
-pub(crate) fn pre_upgrade<T: Config>(version: StorageVersion) -> Result<(), &'static str> {
-	// potentially do some checks on the state before the migration
-	if version < 6 {
-		v6::pre_upgrade::<T>()?;
-	}
-	Ok(())
-}
-
-#[cfg(feature = "try-runtime")]
-pub(crate) fn post_upgrade<T: Config>(version: StorageVersion) -> Result<(), &'static str> {
-	// do some post-migration checks
-	if version < 6 {
-		v6::post_upgrade::<T>()?;
-	}
-	Ok(())
-}
-
 pub(crate) fn migrate<T: Config>() -> Weight {
 	let version = StorageVersion::get::<Pallet<T>>();
 	let mut weight: Weight = Weight::zero();
 
-	if version < 1 {
-		weight.saturating_accrue(v1::migrate::<T>());
-		StorageVersion::new(1).put::<Pallet<T>>();
-	}
-	if version < 2 {
-		weight.saturating_accrue(v2::migrate::<T>());
-		StorageVersion::new(2).put::<Pallet<T>>();
-	}
-	if version < 3 {
-		weight.saturating_accrue(v3::migrate::<T>());
-		StorageVersion::new(3).put::<Pallet<T>>();
-	}
-	if version < 4 {
-		weight.saturating_accrue(v4::migrate::<T>());
-		StorageVersion::new(4).put::<Pallet<T>>();
-	}
-	if version < 5 {
-		weight.saturating_accrue(v5::migrate::<T>());
-		StorageVersion::new(5).put::<Pallet<T>>();
-	}
-	if version < 6 {
-		weight.saturating_accrue(v6::migrate::<T>());
-		StorageVersion::new(6).put::<Pallet<T>>();
-	}
-	if version < 7 {
-		weight.saturating_accrue(v7::migrate::<T>());
-		StorageVersion::new(7).put::<Pallet<T>>();
+	let callbacks: &[&dyn Migrate] = &[
+		&v1::Migration::<T>::new(),
+		&v2::Migration::<T>::new(),
+		&v3::Migration::<T>::new(),
+		&v4::Migration::<T>::new(),
+		&v5::Migration::<T>::new(),
+		&v6::Migration::<T>::new(),
+		&v7::Migration::<T>::new(),
+	];
+
+	for (idx, &calls) in callbacks.iter().enumerate() {
+		let migration_idx = (idx + 1).unique_saturated_into();
+		if version < migration_idx {
+			#[cfg(feature = "try-runtime")]
+			calls.pre_upgrade();
+			weight.saturating_accrue(calls.migrate());
+			StorageVersion::new(migration_idx).put::<Pallet<T>>();
+			#[cfg(feature = "try-runtime")]
+			calls.post_upgrade();
+		}
```

Comment:
Looks good, we can see both v6 and v7 in the logs here https://github.com/gluwa/creditcoin/actions/runs/3940108310/jobs/6741258719 (that's from #816).